src/IOExtras.jl

```diff
@@ -66,14 +66,37 @@ _doc = """
 
 Signal start/end of write or read operations.
 """
-"$_doc"
-startwrite(io) = nothing
-"$_doc"
-closewrite(io) = nothing
-"$_doc"
-startread(io) = nothing
-"$_doc"
-closeread(io) = nothing
+if isdefined(Base, :startwrite)
+    "$_doc"
+    Base.startwrite(io) = nothing
```

Comment:
This is type piracy now

Reply:
Yeah, we'll need to make a plan around this since Base doesn't define a fallback for `closewrite`. Either we add `closewrite(io) = nothing` to Base and we can check if it's defined here, or it's also possible we don't need the fallback anymore. I left this as-is for now since it ensures non-breaking.

Reply:
Because then that's breaking; we have all these code examples all over the package where we tell people to write streaming servers doing `closewrite(stream)`.

Reply:
You can still export it. Downstreams users just must then decide if they want to use `Base.closewrite` or `HTTP.closewrite`.

Reply:
Personally, I would strongly prefer that we remove the type piracy and just make a breaking release of HTTP.jl.

Reply:
https://github.com/JuliaWeb/HTTP.jl/pull/749